lib/SILOptimizer/Utils/LoadStoreOptUtils.cpp

```diff
@@ -245,28 +245,60 @@ void LSLocation::reduce(LSLocation Base, SILModule *M,
     replaceSubLocations(Base, M, context, Locs, SubLocations);
 }
 
-void LSLocation::enumerateLSLocation(TypeExpansionContext context, SILModule *M,
+std::pair<SILValue, bool> LSLocation::getBaseAddress(SILValue v,
```

Comment:
@eeckstein could you use `AccessBase::compute(address)`, `AccessBase::isObjectAccess()`, `AccessBase::getReference()` (then check for immutability), and `AccessBase::getStorageReferenceRoot()` rather than using ad-hoc stripWhateverProjection. This way the possible patterns of object/variable access are well-defined by SIL, verified by the SIL verifier, and the logic used by passes is complete and maintainable.

Note that this version of `getBaseAddress` is misleading because it does not always return a "base address" as defined in MemAccessUtils.

Reply:
Unfortunately it's not that simple. RLE and DSE internally rely on that kind of stripX pattern. When I use AccessBase, I get a crash.
I don't wan't to rewrite and RLE and DSE for this small change - not yet 🙂.

> Note that this version of getBaseAddress is misleading 

I renamed it

Reply:
Was the crash inside an AccessBase method? I'd like to fix that before something else hits it

Reply:
No, it crashed in RLE. There is no problem in AccessBase.